src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace DDTrace\Integrations\Filesystem;
+
+use DDTrace\HookData;
+use DDTrace\Integrations\Integration;
+use DDTrace\Tag;
+
+class FilesystemIntegration extends Integration
+{
+    const NAME = "filesystem";
+
+    public function init(): int
+    {
+        if (!\dd_trace_env_config("DD_APPSEC_RASP_ENABLED")) {
+            return Integration::LOADED;
+        }
+
+        if (!function_exists('datadog\appsec\push_address')) {
+            //Dont load Appsec wrappers is not available
+            return Integration::LOADED;
+        }
+
+        \DDTrace\install_hook(
+            'file_get_contents',
+            self::preHook('file_get_contents'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'file_put_contents',
+            self::preHook('file_put_contents'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'fopen',
+            self::preHook('fopen'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'readfile',
+            self::preHook('readfile'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'stat',
+            self::preHook('stat'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'lstat',
+            self::preHook('lstat'),
+            null
+        );
```

Comment:
We can skip `stat` and `lstat` as I don't think these would be relevant targets for LFI.